### PR TITLE
chore: update CHANGELOG for 202204 release

### PR DESCRIPTION
**Pull request by scotthart**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8684)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -72,6 +74,9 @@ https://github.com/googleapis/google-cloud-cpp/issues/8234.
   [IAM conditions][iam-conditions-link]. They did not work with IAM policies
```
Comment (coryan) [inline CHANGELOG.md:74]: The PRs implementing this change appear multiple times below (#8667, #8652). It gets repetitive. Consider keeping this comment and removing the references to the PRs below.

File: CHANGELOG.md [hunk 2]
```diff
@@ -95,6 +129,38 @@ The library has been expanded to include the following services:
...
+
+### [Common Libraries](https://github.com/googleapis/google-cloud-cpp/blob/main/google/cloud/README.md)
+
+**BREAKING CHANGES**
+
+* cleanup(common)!: remove legacy IAM classes ([#8667](https://github.com/googleapis/google-cloud-cpp/pull/8667))
+
+**Other Changes**
+
+* fix(pubsublite): internal namespaces ([#8680](https://github.com/googleapis/google-cloud-cpp/pull/8680))
```
Comment (coryan) [inline CHANGELOG.md:152]: Just omit all the comments about `pubsublite`

Comment (devbww) [inline CHANGELOG.md:152]: Should we just remove everything about pubsublite?  Or should it at least go under the (previously used) "Pub/Sub Lite" banner?

File: CHANGELOG.md (hunk 1 above)
Comment (dbolduc) [reply CHANGELOG.md:74]: +1

File: CHANGELOG.md [hunk 3]
```diff
@@ -95,6 +129,38 @@ The library has been expanded to include the following services:
...
+* feat(pubsublite): add async version of `GetTopicPartitions()` ([#8640](https://github.com/googleapis/google-cloud-cpp/pull/8640))
+* feat(pubsublite): implementation of user-facing and helper PSL types ([#8591](https://github.com/googleapis/google-cloud-cpp/pull/8591))
+* feat(pubsublite): alarm registry implementation ([#8575](https://github.com/googleapis/google-cloud-cpp/pull/8575))
+* feat(pubsublite): default routing policy ([#8553](https://github.com/googleapis/google-cloud-cpp/pull/8553))
+* feat(pubsublite): partition publisher implementation ([#8478](https://github.com/googleapis/google-cloud-cpp/pull/8478))
+* feat(pubsublite): single partition publisher boilerplate and `Start`/`Shutdown` implementation ([#8531](https://github.com/googleapis/google-cloud-cpp/pull/8531))
+* feat(pubsublite): service composite implementation ([#8529](https://github.com/googleapis/google-cloud-cpp/pull/8529))
+* feat(common): `g::c::future` support for coroutines ([#8532](https://github.com/googleapis/google-cloud-cpp/pull/8532))
+* fix: treat `kDeadlineExceeded` as permanent error ([#8525](https://github.com/googleapis/google-cloud-cpp/pull/8525))
```
Comment (devbww) [inline CHANGELOG.md:161]: What might this be referring to in the "Common Libraries" section?

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8684)* status: 0 of 1 files reviewed, 3 unresolved discussions (waiting on @coryan, @dbolduc, and @devbww)

___
*[CHANGELOG.md, line 74 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8684#-Mzq-IfLa2_cB_7YIp3w-r1-74:-Mzq7roKEh3kcPcpIVEW:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/eee77554096dd2db2f8399e23ec5501b8662affb/CHANGELOG.md#L74)):*
<details><summary><i>Previously, dbolduc (Darren Bolduc) wrote…</i></summary><blockquote>

+1
</blockquote></details>

Done.
___
*[CHANGELOG.md, line 152 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8684#-Mzq-IfLa2_cB_7YIp3w-r1-152:-Mzq7yr71Y3O__ptW87q:bl0wzm2) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/eee77554096dd2db2f8399e23ec5501b8662affb/CHANGELOG.md#L152)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

Should we just remove everything about pubsublite?  Or should it at least go under the (previously used) "Pub/Sub Lite" banner?
</blockquote></details>

Removed all pubsublite commit messages
___
*[CHANGELOG.md, line 161 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8684#-Mzq-IfLa2_cB_7YIp3w-r1-161:-Mzq8Iwf5um9NL182Cy5:bmhkz5g) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/eee77554096dd2db2f8399e23ec5501b8662affb/CHANGELOG.md#L161)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

What might this be referring to in the "Common Libraries" section?
</blockquote></details>

Moved kDeadlineExceeded change to summary section.

**Comment by scotthart** [comment]
> The PR says, "I did not change all of them", so perhaps we need to be a little nuanced? I'm not sure as that may make it less clear.

enumerated services not affected

**Review by coryan — APPROVED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -79,7 +79,11 @@ https://github.com/googleapis/google-cloud-cpp/issues/8234.
 
 **Other Changes**
 
-* fix: treat `kDeadlineExceeded` as permanent error in all services ([#8525](https://github.com/googleapis/google-cloud-cpp/pull/8525))
+* fix: treat `kDeadlineExceeded` as permanent error in all services except ([#8525](https://github.com/googleapis/google-cloud-cpp/pull/8525)):
+  * BigQuery
+  * Bigtable Admin
+  * IAM
+  * Spanner Admin
```
Comment (dbolduc) [inline CHANGELOG.md:86]: All of these services were affected by the PR: https://github.com/googleapis/google-cloud-cpp/pull/8525/commits/7cd327f8ed4d789b991a00a7c459585f6d36c407

I would put it back to the way it was one commit ago.

Comment (devbww) [reply CHANGELOG.md:86]: Sorry about that, Scott.  Perhaps, then, we should be citing two PRs?

Comment (scotthart) [reply CHANGELOG.md]: changed back

**Review by devbww — APPROVED** [review]

**Pull request merged**